File: squadragram-frontend/src/shared/types/character.ts
export type CharacterRole = 'DAMAGE' | 'TANK' | 'TECHNICAL'

export interface Character {
  id: number
  uuid: string
  name: string
  description: string
  role: CharacterRole
}

export interface CharacterFilter {
  role?: CharacterRole;
  name?: string;
}

File: squadragram-backend/internal/model/character.go
package model

import "github.com/google/uuid"

type Character struct {
	ID          int           `json:"id"`
	UUID        uuid.UUID     `json:"uuid"`
	Name        string        `json:"name"`
	Description string        `json:"description"`
	Role        CharacterRole `json:"role"`
}

type CreateCharacterDTO struct {
	Name        string        `json:"name" validate:"required"`
	Description string        `json:"description" validate:"required"`
	Role        CharacterRole `json:"role" validate:"required"`
}
